sysdeps/linux/generic/sysdeps.cpp

```diff
@@ -625,14 +718,206 @@ int sys_reboot(int cmd) {
+
+int sys_epoll_pwait(int epfd, struct epoll_event *ev, int n, int timeout, const sigset_t *sigmask, int *raised) {
+	auto ret = do_syscall(SYS_epoll_pwait, epfd, ev, n, timeout, sigmask, NSIG / 8);
+	if (int e = sc_error(ret); e)
+		return e;
+	*raised = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_eventfd_create(unsigned int initval, int flags, int *fd) {
+	auto ret = do_syscall(SYS_eventfd2, initval, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	*fd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_signalfd_create(const sigset_t *masks, int flags, int *fd) {
+	auto ret = do_syscall(SYS_signalfd4, *fd, masks, 65/8, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	*fd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_timerfd_create(int clockid, int flags, int *fd) {
+	auto ret = do_syscall(SYS_timerfd_create, clockid, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	*fd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_timerfd_settime(int fd, int flags, const struct itimerspec *value, const struct itimerspec *oldvalue) {
+	auto ret = do_syscall(SYS_timerfd_settime, fd, flags, value, oldvalue);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_inotify_create(int flags, int *fd) {
+	sc_result_t ret;
+
+	if(flags) {
+		ret = do_syscall(SYS_inotify_init1, flags);
+	} else {
+		ret = do_syscall(SYS_inotify_init);
+	}
+
+	if (int e = sc_error(ret); e)
+		return e;
+	*fd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_init_module(void *module, unsigned long length, const char *args) {
+	auto ret = do_syscall(SYS_init_module, module, length, args);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_delete_module(const char *name, unsigned flags) {
+	auto ret = do_syscall(SYS_delete_module, name, flags);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_syslog(int type, char *bufp, int len, int *out) {
+	auto ret = do_syscall(SYS_syslog, type, bufp, len);
+	if (int e = sc_error(ret); e)
+		return e;
+	*out = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_socketpair(int domain, int type_and_flags, int proto, int *fds) {
+	auto ret = do_syscall(SYS_socketpair, domain, type_and_flags, proto, fds, 0, 0);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_getsockopt(int fd, int layer, int number, void *__restrict buffer, socklen_t *__restrict size) {
+	auto ret = do_syscall(SYS_getsockopt, fd, layer, number, buffer, size, 0);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_inotify_add_watch(int ifd, const char *path, uint32_t mask, int *wd) {
+	auto ret = do_syscall(SYS_inotify_add_watch, ifd, path, mask);
+	if (int e = sc_error(ret); e)
+		return e;
+	*wd = sc_int_result<int>(ret);
+	return 0;
+}
+
+int sys_inotify_rm_watch(int ifd, int wd) {
+	auto ret = do_syscall(SYS_inotify_rm_watch, ifd, wd);
+	if (int e = sc_error(ret); e)
+		return e;
+	return 0;
+}
+
+int sys_ttyname(int fd, char *buf, size_t size) {
```

Comment:
This sysdep doesn't actually perform any raw syscalls, so I wonder if we could move it to `options/`? The only linux-specific behaviour here is assuming that `/proc/self/fd` is available, but we also rely on `procfs` in some other areas don't we?

Reply:
The `procfs` way is presumably linux-specific and other sysdeps do implement this as a syscall, so moving this into `options/` doesn't look reasonable to me.

Reply:
That is not an entirely satisfying answer, but okay.